Configuration Profiles documentation (#738)
awslabs/aws-advanced-jdbc-wrapper: docs/using-the-jdbc-driver/UsingTheJdbcDriver.md

| `user`                          | `String`  | No       | Database username.                                                                                                                                                                                                                                                            | `null`        |
| `password`                      | `String`  | No       | Database password.                                                                                                                                                                                                                                                            | `null`        |
| `wrapperDialect`                | `String`  | No       | Please see [this page on database dialects](/docs/using-the-jdbc-driver/DatabaseDialects.md), and whether you should include it.                                                                                                                                              | `null`        |

> | `wrapperLogUnclosedConnections` | `Boolean` | No       | Allows the AWS JDBC Driver to capture a stacktrace at each connection opening and if the `finalize()` method is reached without having been closed the stacktrace is printed to the log. It helps developers to detect and correct the source of pothential connection leaks. | `false`       |

Reviewer: ```suggestion
| `wrapperLogUnclosedConnections` | `Boolean` | No       | Allows the AWS JDBC Driver to capture a stacktrace for each connection that is opened. If the `finalize()` method is reached without the connection being closed, the stacktrace is printed to the log. This helps developers to detect and correct the source of potential connection leaks. | `false`       |
```
?

- [Target Driver Dialect](./using-the-jdbc-driver/TargetDriverDialects.md#target-driver-dialects)
- a custom exception handler
- a custom connection provider

The following example creates and sets a configuration profile:

```java
// Create a new configuration profile with name "testProfile"
ConfigurationProfileBuilder.get()
    .withName("testProfile")
    .withPluginFactories(Arrays.asList(
        FailoverConnectionPluginFactory.class,
        HostMonitoringConnectionPluginFactory.class,
        CustomConnectionPluginFactory.class))
    .buildAndSet();

// Use of configuration profile "testProfile"
properties.setProperty("wrapperProfileName", "testProfile");
```

Configuration profile can be created based on another existing configuration profile. Profile names should be unique (case sensitive) and no duplication is allowed.

```java
// Create a new configuration profile with name "newProfile" based on "existingProfileName"
ConfigurationProfileBuilder.from("existingProfileName")
    .withName("newProfileName")
    .withDialect(new CustomDatabaseDialect())
.buildAndSet();

// Delete configuration profile "testProfile"
DriverConfigurationProfiles.remove("testProfile");
```

> AWS JDBC Driver team has gathered and analyzed various user scenarios and created configuration presets for you. Such preset configuration profiles are optimized, profiled and verified and could be used right away. Preset configurations are the same as configuration profiles explained above. Users can create their own configuration profiles based on built-in presets as shown above. More details could be found at [Configuration Presets](./ConfigurationPresets.md) page.  

Reviewer: ```suggestion
The AWS JDBC Driver team has gathered and analyzed various user scenarios to create commonly used configuration profiles, or presets, for users. These preset configuration profiles are optimized, profiled, verified and can be used right away. Users can create their own configuration profiles based on the built-in presets as shown above. More details could be found at the [Configuration Presets](./ConfigurationPresets.md) page.  
```
